File: libs/eave-stdlib-ts/src/core-api/operations/slack.ts
import { sharedConfig } from '../../config.js';
import { EaveApp } from '../../eave-origins.js';
import { CtxArg, RequestArgsOrigin, makeRequest } from '../../requests.js';
import { SlackInstallation, SlackInstallationInput } from '../models/slack.js';
import { Team } from '../models/team.js';
import { CoreApiEndpointConfiguration } from './shared.js';

export type GetSlackInstallationRequestBody = {
  slack_integration: SlackInstallationInput;
}

export type GetSlackInstallationResponseBody = {
  team: Team;
  slack_integration: SlackInstallation;
}

export class GetSlackInstallationOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/integrations/slack/query" })

  static async perform(args: RequestArgsOrigin & {input: GetSlackInstallationRequestBody}): Promise<GetSlackInstallationResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <GetSlackInstallationResponseBody>(await resp.json());
    return responseData;
  }
}

File: libs/eave-stdlib-py/src/eave/stdlib/core_api/operations/slack.py
from typing import Unpack
from eave.stdlib.api_types import BaseRequestBody, BaseResponseBody
from eave.stdlib.core_api.models.slack import SlackInstallation
from eave.stdlib.core_api.models.slack import SlackInstallationInput
from . import CoreApiEndpoint, CoreApiEndpointConfiguration

from ..models import team
from ... import requests


class GetSlackInstallation(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/integrations/slack/query",
        auth_required=False,
        team_id_required=False,
    )

    class RequestBody(BaseRequestBody):
        slack_integration: SlackInstallationInput

    class ResponseBody(BaseResponseBody):
        team: team.Team
        slack_integration: SlackInstallation

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=input,
            **kwargs,
        )
